### PR TITLE
OCPBUGS-24612: check for goexperimental >= 1.18

### LINKED ISSUE #81: go: binary has zero tags enabled (should have strictfipsruntime)
If the GOEXPERIMENT=strictfipsruntime is used, the check-payload, there is no tag for strictfipsruntime created, and the check-payload tool complains about `go: binary has zero tags enabled (should have strictfipsruntime)`

### PR DESCRIPTION
**Pull request by rphillips**
Check for goexperimental with golang >= 1.18.

Fixed #81

### CODE REVIEW

**Comment by rphillips** [comment]
This pr is ready. Passes on 4.14+.

**Comment by joepvd** [comment]
Think the behavior with this change is:
| -tags=strictfipsruntime | GOEXPERIMENT=strictfipsruntime | result |
| ----------------------- | ------------------------------ | ------ |
| yes                     | yes                            | good   |
| yes                     | no                             | bad    |
| no                      | yes                            | bad    |
| no                      | no                             | bad    |

I think the behavior should be:

| -tags=strictfipsruntime | GOEXPERIMENT=strictfipsruntime | result |
| ----------------------- | ------------------------------ | ------ |
| yes                     | yes                            | good   |
| yes                     | no                             | good   |
| no                      | yes                            | good   |
| no                      | no                             | bad    |

**Comment by rphillips** [comment]
Thanks. Updated the PR.

**Review by joepvd — APPROVED** [review]
Nice, thanks!

/lgtm

File: internal/validations/validations.go [hunk 1]
```diff
@@ -107,37 +107,57 @@ func validateGoCgo(_ context.Context, _ string, baton *Baton) *types.ValidationE
...
+
+	// check for GOEXPERIMENT``
+	expResult := func() *types.ValidationError {
+		for _, bs := range baton.GoBuildInfo.Settings {
+			if bs.Key == "GOEXPERIMENT" && !strings.Contains(bs.Value, "strictfipsruntime") {
+				return types.NewValidationError(types.ErrGoNotGoExperiment)
+			}
+		}
+		return nil
+	}
```
Comment (dollierp) [inline internal/validations/validations.go:156]: Missing parentheses after bracket?: `}()`

**Pull request merged**